vllm/v1/engine/output_processor.py

```diff
@@ -201,18 +201,37 @@ def abort_requests(
         self,
         request_ids: Iterable[str],
     ) -> list[str]:
-        request_ids_to_abort = []
-        for request_id in request_ids:
-            req_state = self.request_states.pop(request_id, None)
+        request_ids_to_abort = self.flatten_req_to_abort(request_ids)
+        self.handle_abort_reqs(request_ids_to_abort)
+        return request_ids_to_abort
+
+    def flatten_req_to_abort(self, req_ids: Iterable[str]) -> list[str]:
+        ret = []
+        for req_id in req_ids:
+            parent = self.parent_requests.pop(req_id, None)
+            if parent is None:
+                ret.append(req_id)
+            else:
+                ret.extend(parent.child_requests)
+        return ret
+
+    # "Aborted request", meaning the frontend first detects that
+    # the request has ended, such as when the client disconnects
+    # or the detokenizer detects a stop string.
+    def handle_abort_reqs(self, req_ids: Iterable[str]):
+        for req_id in req_ids:
+            req_state = self.request_states.pop(req_id, None)
             if req_state is not None:
                 self.lora_states.abort_request(req_state)
-                request_ids_to_abort.append(request_id)
-            else:
-                parent = self.parent_requests.pop(request_id, None)
-                if parent and parent.child_requests:
-                    self.abort_requests(parent.child_requests)
-                    request_ids_to_abort.extend(parent.child_requests)
-        return request_ids_to_abort
+        return
+
+    # "Finished request", meaning EngineCore first detects that
+    # the request has ended, and the resources related to the request
+    # maintained by EngineCore have been released.
+    def _handle_finished_reqs(self, req_id):
```

Comment:
lets call this `def finish_request(self, request_id: str) -> None`

Reply:
done